pkg/virtctl/imageupload/imageupload.go

```diff
@@ -926,3 +936,64 @@ func handleEventErrors(client kubecli.KubevirtClient, pvcName, dvName, namespace
 
 	return nil
 }
+
+func createDataSource(client kubecli.KubevirtClient, name, namespace string) error {
```

Comment:
Can you split this func? It's quite long. Also, it does not only create a DS, it can also patch an existing DS.

Reply:
`patchOrCreateDataSource` -> calls `createDataSource` or `patchDataSource`

Reply:
Yes I'll split it.